packages/trader/src/AppV2/Hooks/useContractsForCompany.ts

```diff
@@ -0,0 +1,86 @@
+
+type TContractType = {
+    text?: string;
+    value: string;
+};
+
+export type TContractTypesList = {
+    [key: string]: {
+        name: string;
+        categories: DeepRequired<TContractType[]>;
+    };
+};
+
+const useContractsForCompany = () => {
+    const [contract_types_list, setContractTypesList] = React.useState<TContractTypesList | []>([]);
+    const { setContractTypesListV2 } = useTraderStore();
+    const { client } = useStore();
+    const { landing_company_shortcode } = client;
+    const contract_categories = getContractCategoriesConfig();
+    const available_categories = cloneObject(contract_categories);
+    const contract_types = getContractTypesConfig();
+    let available_contract_types: ReturnType<typeof getContractTypesConfig> = {};
+
+    const fetchContractForCompany = React.useCallback(async () => {
+        let response;
+
+        const request = {
+            landing_company: landing_company_shortcode,
+        };
+
+        try {
+            response = await WS.contractsForCompany(request);
+            const { contracts_for_company = [], error } = response;
+            available_contract_types = {};
+
+            if (!error && contracts_for_company?.available.length) {
+                contracts_for_company.available.forEach((contract: any) => {
+                    const type = Object.keys(contract_types).find(
+                        key =>
+                            contract_types[key].trade_types.indexOf(contract.contract_type) !== -1 &&
+                            (contract.contract_type !== 'PUT' || contract_types[key].barrier_count === 1) // To distinguish betweeen Rise/Fall & Higher/Lower
+                    );
+
+                    if (!type) return; // ignore unsupported contract types
+
+                    if (!available_contract_types[type]) {
+                        // extend contract_categories to include what is needed to create the contract list
+                        const sub_cats =
+                            available_categories[
+                                Object.keys(available_categories).find(
+                                    key => available_categories[key].categories.indexOf(type) !== -1
+                                ) ?? ''
+                            ].categories;
+
+                        if (!sub_cats) return;
+
+                        sub_cats[(sub_cats as string[]).indexOf(type)] = {
+                            value: type,
+                            text: contract_types[type].title,
+                        };
+
+                        available_contract_types[type] = cloneObject(contract_types[type]);
+                    }
+                });
+
+                setContractTypesListV2(available_categories);
+                setContractTypesList(available_categories);
+            }
+        } catch (err) {
+            /* eslint-disable no-console */
+            console.error(err);
```

Comment:
I don't remember if anyone asked: do we want to get this error in the console and TrackJS?
Perhaps, it's better to set the error instead?

```suggestion
            setError(true, err);
```

Reply:
There're actually no requirements to display error message to users as of now, so I just logged it in the catch block to be aware in case something unpredictable happens. I'm not sure what to do with it, but I think it's not bad to have this error in trackJs, at least we can see if something is not going well with API response. What do you think?

Reply:
i can see code in the code base but with console.dir, I guess we can leave it for now. In case we start getting new trackjs errors after release, we'll revisit this👍🏻

Reply:
Well, the point of the log is to alert about the issues, if we start getting trackjs errors, we need to identify the reason and fix the logic, no matter where it is, and not to remove the log 😄